internal/k8s/pod/pod.go

```diff
@@ -0,0 +1,303 @@
+	}
+	return mgr, nil
+}
+
+func (pi *podInformer) indexerFunc(obj client.Object) []string {
+	pod, ok := obj.(*corev1.Pod)
+	if !ok {
+		return []string{"invalidContainerID"}
+	}
+	var containerIDs []string
+	// TODO: check for status.State.Running != nil ?
+	for _, status := range pod.Status.ContainerStatuses {
+		if status.ContainerID != "" {
+			containerIDs = append(containerIDs, extractContainerID(status.ContainerID))
+		}
+	}
+	for _, status := range pod.Status.EphemeralContainerStatuses {
+		if status.ContainerID != "" {
+			containerIDs = append(containerIDs, extractContainerID(status.ContainerID))
+		}
+	}
+	for _, status := range pod.Status.InitContainerStatuses {
+		if status.ContainerID != "" {
+			containerIDs = append(containerIDs, extractContainerID(status.ContainerID))
+		}
+	}
+	pi.logger.Debug(
+		"containers for pod",
+		"pod",
+		pod.Name,
+		"containers",
+		strings.Join(containerIDs, ","),
+	)
+	return containerIDs
+}
+
+func extractContainerID(str string) string {
+	parts := strings.SplitN(str, "://", 2)
+	return parts[len(parts)-1]
+}
+
+func (pi *podInformer) Run(ctx context.Context) error {
+	pi.logger.Info("Starting pod informer")
+	mgrCtx, cancel := context.WithCancel(context.Background())
+	pi.mgrCtx = mgrCtx
+	pi.mgrCancelFunc = cancel
+	return pi.manager.Start(mgrCtx)
+}
+
+// PodInfo retrieves pod details given a containerID
+func (pi *podInformer) PodInfo(containerID string) (*PodInfo, error) {
+	var pods corev1.PodList
+
+	err := pi.manager.GetCache().List(
+		context.Background(),
+		&pods,
+		client.MatchingFields{IndexContainerID: containerID},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving pod info from cache: %w", err)
+	}
+
+	if len(pods.Items) == 0 {
+		// pi.logger.Debug("no pod found for container", "container", containerID)
+		return nil, ErrNotFound
+	}
+	if len(pods.Items) > 1 {
+		return nil, fmt.Errorf("multiple pods found for containerID: %s", containerID)
+	}
+
+	pod := pods.Items[0]
+	pi.logger.Debug("pod found for container", "container", containerID, "pod", pod.Name)
+	return &PodInfo{
+		ID:        string(pod.UID),
+		Name:      pod.Name,
+		Namespace: pod.Namespace,
+	}, nil
+}
+
+func getConfig(kubeConfigPath string) (*rest.Config, error) {
+	return clientcmd.BuildConfigFromFlags("", kubeConfigPath)
+}
+
+func (i *podInformer) Name() string {
+	return "podInformer"
+}
+
+func (pi *podInformer) Shutdown() error {
+	pi.logger.Info("stopping pod informer manager")
+	pi.mgrCancelFunc()
+	return nil
+}
+
+type dummyReconciler struct{}
+
+func (pi *dummyReconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	return reconcile.Result{}, nil
+}
+
+func createController(mgr manager.Manager) error {
+	d := &dummyReconciler{}
+	_, err := ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.Pod{}).
+		WithEventFilter(predicate.ResourceVersionChangedPredicate{}).
+		Build(d)
+	return err
```

Comment:
we shouldn't need the reconciler since manager's cache is all that we need - right?